### ISSUE TITLE
Provide managed APIs for our canonical set of runtime performance counters

### ISSUE THREAD

**Issue by tommcdon** [dotnet/runtime#36571]
[AB#1117223](https://devdiv.visualstudio.com/10e66e43-9645-4201-b128-0fdc3769cc17/_workitems/edit/1117223)
Provide the data needed to build managed performance counters

**Comment by tommcdon** [comment]
Depends on #34648, #31951, #1099, #1098

**Comment by tommcdon** [comment]
@sywhang do you have a PR for this issue?

**Comment by sywhang** [comment]
There are issues I filed for API reviews. Some of them were closed because there are ways to compute them using the managed APIs:

https://github.com/dotnet/runtime/issues/36324 (JIT metrics): this one has some discussions going on regarding what metrics to expose. I think it might be better to hold off on this for 6.0 so that we can expose all the right metrics instead of adding just a couple of ones that we are currently using so that we have a more complete story, unless you think we should get it into 5.0?

https://github.com/dotnet/runtime/issues/36068: there is a way to get this using managed API already. 

https://github.com/dotnet/runtime/issues/36071: same as above

https://github.com/dotnet/runtime/issues/36069: same as above.
